### ISSUE TITLE
Wasm contracts duplicated in the state

### ISSUE THREAD

**Issue by herr-seppia** [dusk-network/rusk#575]
**Describe the bug**
The content of the wasm contracts seems to be written in the state each time that the latter is persisted

**To Reproduce**

Restore a genesis state with `target/release/rusk-recovery-state -f`
```
    Checking Network state
 Downloading state from previous build
   Unzipping state archive into
      Stored network state at <STATE_FOLDER>
      Stored persisted id at <STATE_ID>
    Finished task in 0.46s
```

Check the size of `<STATE_FOLDER>/data` file (mine is 552812)

In `rusk/test/services/transaction.rs` replace 
```
 fn testbackend() -> BackendCtor<DiskBackend> {
    BackendCtor::new(DiskBackend::ephemeral)
 }
```
with 
```
pub fn ephemeral() -> Result<DiskBackend, PersistError> {
    let backend = DiskBackend::ephemeral()?;
    info!("debug path {:?}", backend);
    Ok(backend)
}

 fn testbackend() -> BackendCtor<DiskBackend> {
    BackendCtor::new(ephemeral)
 }
```

Launch the grpc_wallet test `cargo t --release wallet_grpc` and note down the folder of the state
The test will persist the state twice, and the size of the new state will be 3*<original_size>

Furthermore the transfer contract wasm contains the string "unicode" twice
`grep -c unicode target/wasm32-unknown-unknown/release/transfer_contract.wasm` --> 2

The stake contract contains the string "unicode" twice too
`grep -c unicode target/wasm32-unknown-unknown/release/stake_contract.wasm` --> 2

In the inital (genesis state) we have 4 "unicode" 
`grep -c unicode <STATE_FOLDER>/data` --> 4

After the grpc_wallet test, (that create 2 additional blocks) I have 12 "unicode"

In the cluster, after reach the block 18079 I have exactly 72320 "unicode" (4+18079*4)

**Additional context**
See also #563

**Comment by herr-seppia** [comment]
I found a way to distinct between stake and transfer duplication

`grep -c contracts/transfer target/wasm32-unknown-unknown/release/transfer_contract.wasm` --> 2
`grep -c contracts/stake target/wasm32-unknown-unknown/release/stake_contract.wasm`--> 1

After 2 block I found that
`grep -c contracts/transfer /var/folders/s2/j876kzjd1gj4_mlb61q_pwpc0000gn/T/.tmpI9uTLA/data` --> 6
`grep -c contracts/stake /var/folders/s2/j876kzjd1gj4_mlb61q_pwpc0000gn/T/.tmpI9uTLA/data` --> 3

**Comment by autholykos** [comment]
This is fixed by #609
